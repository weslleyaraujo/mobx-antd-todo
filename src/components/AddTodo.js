// @flow
import React, { Fragment } from 'react';
import yup from 'yup';
import { Box } from 'grid-styled';
import { Formik } from 'formik';
import { Input } from 'antd';
import { Layout, Col, Row, Card, Button, Tooltip, List, Tag } from 'antd';

const schema = yup.object().shape({
  text: yup
    .string()
    .min(4, 'Too short.')
    .required('Required.')
});

function AddTodo({ onSubmit }: { onSubmit: (...args: any) => void }) {
  return (
    <Formik
      initialValues={{ text: '' }}
      validationSchema={schema}
      onSubmit={onSubmit}
      render={({
        values,
        errors,
        touched,
        handleChange,
        handleBlur,
        handleSubmit,
        isSubmitting,
        isValid
      }) => (
        <Fragment>
          <form onSubmit={handleSubmit}>
            <div className={errors.text ? 'has-error' : ''}>
              <Input
                value={values.text}
                onChange={handleChange}
                name="text"
                onBlur={handleBlur}
              />
              {touched.text &&
              errors.text && (
                <div className="ant-form-explain">{errors.text}</div>
              )}
            </div>
            <Row justify="end" type="flex">
              <Col>
                <Box py={3}>
                  <Button
                    htmlType="submit"
                    type="primary"
                    disabled={isSubmitting || !isValid}
                    loading={isSubmitting}
                  >
                    Add todo
                  </Button>
                </Box>
              </Col>
            </Row>
          </form>
        </Fragment>
      )}
    />
  );
}

export default AddTodo;
