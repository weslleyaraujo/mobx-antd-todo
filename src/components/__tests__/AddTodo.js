import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import AddTodo from "../AddTodo";

it("renders AddTodo correctly", () => {
  const wrapper = shallow(<AddTodo />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
