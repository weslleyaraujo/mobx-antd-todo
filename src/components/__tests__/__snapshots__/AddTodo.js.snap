// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddTodo correctly 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "text": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": false,
      "_nodes": Array [
        "text",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "text": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "min": true,
            "required": false,
          },
          "_mutate": false,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
/>
`;
